src/grid_interpolator.cc

```diff
@@ -0,0 +1,84 @@
+#include "flyft/grid_interpolator.h"
+
+#include <cmath>
+#include <cassert>
+#include <fstream>
+#include <tuple> 
+
+namespace flyft
+{
+GridInterpolator::GridInterpolator(const std::string& s)
+    {
+    std::ifstream indata(s);
+    if(!indata)
+        {
+        throw std::runtime_error("File failed to open");
+        }
+        
+    int nx, ny, nz;   
+    indata >> nx >> ny >> nz >> dx_ >> dy_ >> dz_;
+    n_ = ThreeDimensionalIndex(nx, ny, nz);
+    if (n_.size() == 0)
+        {
+        throw std::runtime_error("Grid has zero size");
+        }
+        
+    data_ = new double[n_.size()];  
+    double value;
+    
+    int idx = 0;
+    while (indata >> value)
+        {
+        data_[idx++] = value;
+        }
+        
+    if (idx != n_.size() || !indata.eof())
+        {
+        throw std::runtime_error("File did not right number of elements");
+        }
+        
+    indata.close();
+    }
+
+GridInterpolator::~GridInterpolator()
+    {
+    delete[] data_;
+    }
+
+double GridInterpolator::operator()(double x, double y, double z) const
+    {
+    const double fx = x/dx_;
+    const double fy = y/dy_;
+    const double fz = z/dz_;
+    const int bin_x = std::floor(fx); 
+    const int bin_y = std::floor(fy); 
+    const int bin_z = std::floor(fz); 
+    const double xd = fx - bin_x;
+    const double yd = fy - bin_y;
+    const double zd = fz - bin_z;
+    #ifndef NDEBUG
+    const auto bounds = n_.getBounds();
+    assert(bin_x >= 0 && bin_x < std::get<0>(bounds));
+    assert(bin_y >= 0 && bin_y < std::get<1>(bounds));
+    assert(bin_z >= 0 && bin_z < std::get<2>(bounds));
+    #endif
```

Comment:
whitespace for readability
```suggestion

    #ifndef NDEBUG
    const auto bounds = n_.getBounds();
    assert(bin_x >= 0 && bin_x < std::get<0>(bounds));
    assert(bin_y >= 0 && bin_y < std::get<1>(bounds));
    assert(bin_z >= 0 && bin_z < std::get<2>(bounds));
    #endif

```